### PR TITLE
#3530 Added wishlist check

### PR DESCRIPTION
**Pull request by VanilStorm**
**Related issue(s):**
* Fixes https://github.com/scandipwa/scandipwa/issues/3530

**Problem:**
* Mobile: Share button is active in Wishlist

**In this PR:**
* Added wishlist check

### CODE REVIEW

File: packages/scandipwa/src/component/Header/Header.component.js [hunk 1]
```diff
@@ -325,9 +325,20 @@ export class Header extends NavigationAbstract {
...
-            shareWishlist
+            shareWishlist,
+            navigationState: { title }
         } = this.props;
 
+        if (!title) {
+            return null;
+        }
+
+        const wishListQty = Number(title.split(' ').filter((item) => Number(+item)));
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/Header/Header.component.js:336]: Well, you definitely could find wishlist qty another way.

**Review by AleksandrsKondratjevs — CHANGES_REQUESTED** [review]

File: packages/scandipwa/src/component/Header/Header.component.js [hunk 2]
```diff
@@ -325,9 +326,16 @@ export class Header extends NavigationAbstract {
...
         const {
             isWishlistLoading,
-            shareWishlist
+            shareWishlist,
+            productsInWishlist
         } = this.props;
 
+        const wishListQty = Object.keys(productsInWishlist).length;
+
+        if (!wishListQty) {
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/Header/Header.component.js:335]: perhaps there is no point to create variable for 1 time use in if block condition.

File: packages/scandipwa/src/component/Header/Header.component.js [hunk 3]
```diff
@@ -81,6 +81,7 @@ export class Header extends NavigationAbstract {
             onBackClick: PropTypes.func,
             title: PropTypes.string
         }).isRequired,
+        productsInWishlist: PropTypes.objectOf(PropTypes.number),
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/Header/Header.component.js:84]: I assume those Proptypes are not correct.  All types in object are numbers ? :/

**Review by AleksandrsKondratjevs — CHANGES_REQUESTED** [review]

File: packages/scandipwa/src/component/Header/Header.component.js (hunk 3 above)
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/Header/Header.component.js:84]: Recheck propTypes, I assume they are not numbers. It is array ob products objects.

**Review by AleksandrsKondratjevs — CHANGES_REQUESTED** [review]

File: packages/scandipwa/src/component/Header/Header.component.js (hunk 3 above)
Comment (VanilStorm) [reply packages/scandipwa/src/component/Header/Header.component.js:84]: actually, productsInWishlist is an object that stores other product objects, the key of which is a number
Снимок экрана 2022-03-28 в 14 17 28
.

File: packages/scandipwa/src/component/Header/Header.component.js [hunk 4]
```diff
@@ -81,6 +81,7 @@ export class Header extends NavigationAbstract {
             onBackClick: PropTypes.func,
             title: PropTypes.string
         }).isRequired,
+        productsInWishlist: PropTypes.objectOf(PropTypes.object),
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/Header/Header.component.js:84]: Object is deprecated proptype. Check proptypes in Types same as check where same state is get, maybe you can just reuse already made logic.

**Review by AleksandrsKondratjevs — CHANGES_REQUESTED** [review]

**Comment by VanilStorm** [comment]
Closed this since it has reverted changes and it's easier to fix in a different branch.
See #4616

**Pull request closed without merging**